6.1	Өгөгдсөн а тоог анхны тоо мөн эсэхийг ол.

In [1]:
a = int(input('toog oruul: '))
isprime = True

for i in range(2, int(a/2)):
    if a % i == 0:
        isprime = False

if isprime:
    print('Anhni too mon')
else:
    print('Anhni too bish')

toog oruul: 17
Anhni too mon


6.2	Өгөгдсөн n тоо хичнээн оронтой эсэхийг ол.

In [2]:
a = int(input('toog oruul: '))

print(len(str(a)))

toog oruul: 2568993
7


6.3	Эхний ба сүүлийн цифр ижил байх тоонуудыг хэвлэ.

In [3]:
for i in range(100, 1000):
    if str(i)[0] == str(i)[2]:
        print(i)

101
111
121
131
141
151
161
171
181
191
202
212
222
232
242
252
262
272
282
292
303
313
323
333
343
353
363
373
383
393
404
414
424
434
444
454
464
474
484
494
505
515
525
535
545
555
565
575
585
595
606
616
626
636
646
656
666
676
686
696
707
717
727
737
747
757
767
777
787
797
808
818
828
838
848
858
868
878
888
898
909
919
929
939
949
959
969
979
989
999


6.4	Өгөгдсөн бүхэл тоо палиндром тоо эсэхийг тодорхойлох. Урд, ардаасаа ижил уншигддаг тоог палиндром тоо гэнэ. Жишээ нь: 121, 456654 гэх мэт.

In [4]:
a = 1
while a < 10:
    a = int(input('toog oruul [a>10]: '))

print('palindrome mon' if str(a) == str(a)[::-1] else 'palindrome bish')

toog oruul [a>10]: 4654654
palindrome bish


6.5	2 суурьтай системд өгсөн тоог 10 суурьтай тооллын системд хөрвүүлэх. 

In [5]:
binary = int(input('2tiin too oruul:'))
sum = 0
temp = len(str(binary))-1
for i in str(binary):
    sum += int(i)*(2**temp)
    temp-=1

print(sum)

2tiin too oruul:1001
9


6.6	16 суурьтай системд өгсөн тоог 10 суурьтай тооллын системд хөрвүүлэх. 

In [6]:
hexa = int(input('2tiin too oruul:'), 2)

print(hex(hexa)[2])

2tiin too oruul:1001
9


6.7	Фибоначчийн эхний к ширхэг гишүүнийг хэвлэ (Фибоначчийн тоо нь дараах дарааллаас бүрдэнэ: 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, .... Тодорхойлолт ёсоор бол Фибоначчийн тооны эхний 2 тоо нь 0 болон 1 гэж өгөгдсөн ба түүнээс хойшхи Фибоначчийн тоонууд нь өмнөх 2 тооныхоо нийлбэртэй тэнцүү болно).

In [7]:
k = 0

while k<2:
    k = int(input('Хэдэн ширхэг тоог олохоо оруул(санамж! к тоо нь 2-с их бүхэл тоо байх ёстой): '))

first = 0
second = 1
print('0, 1', end=', ')

for i in range(k-2):
    print(first+second, end=', ')
    temp = second + first
    first = second
    second = temp

Хэдэн ширхэг тоог олохоо оруул(санамж! к тоо нь 2-с их бүхэл тоо байх ёстой): 6
0, 1, 1, 2, 3, 5, 

6.8	Фибоначчийн к-д хуваагдах хамгийн бага гишүүнийг ол.

In [8]:
k = int(input('к тоог олохоо оруул: '))

first = 0
second = 1

while True:
    temp = second + first
    if temp % k == 0:
        print(temp)
        break
    first = second
    second = temp

к тоог олохоо оруул: 61
610


6.9	Фибоначчийн 3 оронтой бүх гишүүдийг хэвлэ.

In [9]:
first = 0
second = 1
temp = 0
while temp<1000:
    if temp > 100:
        print(temp)
    temp = second + first
    first = second
    second = temp

144
233
377
610
987


6.10	[a,b] завсар дах хөрвөсөн тооноосоо их байх тоонуудыг хэвлэ.

In [10]:
a = int(input('a toog oruul: '))
b = int(input('b toog oruul: '))

for i in range(a, b+1):
    if i > int(str(i)[::-1]):
        print(i)

a toog oruul: 10
b toog oruul: 35
10
20
21
30
31
32


6.11	[a,b] завсар дах хөрвөсөн тооноосоо бага байх тоонуудыг хэвлэ.

In [11]:
a = int(input('a toog oruul: '))
b = int(input('b toog oruul: '))

for i in range(a, b+1):
    if i < int(str(i)[::-1]):
        print(i)

a toog oruul: 10
b toog oruul: 35
12
13
14
15
16
17
18
19
23
24
25
26
27
28
29
34
35


6.12	[a,b] завсар дах тоонууд дотроос 
а. К тооны тэгш (сондгой ) цифр агуулсан 
б. Бүх цифр нь тэгш (сондгой ) байх тоонуудыг хэвлэх. 

     a

In [12]:
a = int(input('a toog oruul: '))
b = int(input('b toog oruul: '))
k = int(input('k toog oruul: '))

if a>b:
    temp = b
    b=a
    a=temp

for i in range(a, b+1):
    sum = 0
    for j in str(i):
        if int(j)%2==0:
            sum+=1
    if sum == k:
        print('k tooni tegsh cipr aguulsan:', i)
    if len(str(i))-sum == k:
        print('k tooni sondgoi cipr aguulsan:', i)

a toog oruul: 10
b toog oruul: 35
k toog oruul: 10


     b

In [13]:
a = int(input('a toog oruul: '))
b = int(input('b toog oruul: '))

if a>b:
    temp = b
    b=a
    a=temp

for i in range(a, b+1):
    isEven = True
    isOdd = True
    for j in str(i):
        if int(j) % 2 == 0:
            isOdd = False
            continue
        isEven = False
    for j in str(i):
        if int(j) % 2 == 1:
            isEven = False
            continue
        isOdd = False
    if isEven:
        print('Buh cipr ni tegsh:', i)
    elif isOdd:
        print('Buh cipr ni sondgoi', i)
    

a toog oruul: 10
b toog oruul: 20
Buh cipr ni sondgoi 11
Buh cipr ni sondgoi 13
Buh cipr ni sondgoi 15
Buh cipr ni sondgoi 17
Buh cipr ni sondgoi 19
Buh cipr ni tegsh: 20
